src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java

```diff
@@ -207,27 +206,32 @@ private TransactionClose(String msg) {
     public synchronized  Map<KVQuery, List<KeyValue>> getMultiRange(final List<Object[]> queries)
             throws PermanentBackendException {
         Map<KVQuery, List<KeyValue>> resultMap = new ConcurrentHashMap<>();
-        final List<CompletableFuture> futures = new LinkedList<>();
         try {
-            for(Object[] obj : queries) {
-                final KVQuery query = (KVQuery) obj[0];
-                final byte[] start = (byte[]) obj[1];
-                final byte[] end = (byte[]) obj[2];
-
-                futures.add(tx.getRange(start, end, query.getLimit()).asList()
-                        .whenComplete((res, th) -> {
-                            if (th == null) {
-                                if (res == null) {
-                                    res = Collections.emptyList();
-                                }
-                                resultMap.put(query, res);
-                            } else {
-                                log.error("failed to complete getRange of multiRange", th);
-                                throw new RuntimeException("failed to getRange for multiRangeQuery", th);
-                            }
-                        }));
+            for (int i = 0; i < maxRuns; i++) {
```

Comment:
the new idea is to trigger a completableFuture for each getRange. As multiRange is a list of range queries, number of completableFuture = no of getRange. only retry futures that failed. this eventually means that,  if 1 future failed in the first retry and passed in second retry, the total time taken will be 2x [ x is avg time taken in try ] . Each retry's callabale futres are exdected after all previous callableFuture has completed.